File: order-service/src/main/java/com/ordersystem/orderservice/domain/model/OrderCreatedEvent.java
package com.ordersystem.orderservice.domain.model;

import java.util.UUID;

// Representa el hecho de que una orden fue creada.
// Inmutable por diseño — los eventos no se modifican, solo se publican.
public record OrderCreatedEvent(
        UUID orderId,
        String customerId,
        String productId,
        int quantity
) {}

File: inventory-service/src/inventory_service/domain/model/order_created_event.py
from dataclasses import dataclass
from uuid import UUID


# Espejo del record Java OrderCreatedEvent — representa el hecho de que
# el order-service creó una orden y nosotros debemos verificar el stock.
# Es solo datos: sin lógica, sin imports de infraestructura.
@dataclass
class OrderCreatedEvent:
    order_id: UUID    # identificador único de la orden
    customer_id: str  # quién hizo la orden (para el evento de respuesta)
    product_id: str   # identificador del producto — se mapea contra 'sku' en nuestra DB
    quantity: int     # cuántas unidades pidió el cliente